app/controllers/general.php

```diff
@@ -234,6 +230,61 @@ function router(App $utopia, Database $dbForConsole, SwooleRequest $swooleReques
             Request::setFilter(null);
         }
 
+        $domain = $request->getHostname();
+        $domains = Config::getParam('domains', []);
+        if (!array_key_exists($domain, $domains)) {
+            $domain = new Domain(!empty($domain) ? $domain : '');
+
+            if (empty($domain->get()) || !$domain->isKnown() || $domain->isTest()) {
+                $domains[$domain->get()] = false;
+                Console::warning($domain->get() . ' is not a publicly accessible domain. Skipping SSL certificate generation.');
+            } elseif (str_starts_with($request->getURI(), '/.well-known/acme-challenge')) {
+                Console::warning('Skipping SSL certificates generation on ACME challenge.');
+            } else {
+                Authorization::disable();
+
+                $envDomain = App::getEnv('_APP_DOMAIN', '');
+                $mainDomain = null;
+                if (!empty($envDomain) && $envDomain !== 'localhost') {
+                    $mainDomain = $envDomain;
+                } else {
+                    $domainDocument = $dbForConsole->findOne('rules', [Query::orderAsc('$id')]);
+                    $mainDomain = $domainDocument ? $domainDocument->getAttribute('domain') : $domain->get();
+                }
+
+                if ($mainDomain !== $domain->get()) {
+                    Console::warning($domain->get() . ' is not a main domain. Skipping SSL certificate generation.');
+                } else {
+                    $domainDocument = $dbForConsole->findOne('rules', [
+                        Query::equal('domain', [$domain->get()])
+                    ]);
+
+                    if (!$domainDocument) {
+                        $domainDocument = new Document([
+                            'domain' => $domain->get(),
+                            'resourceType' => 'api',
+                            'status' => 'verifying',
+                            'projectId' => 'console',
+                            'projectInternalId' => 'console'
+                        ]);
+
+                        $domainDocument = $dbForConsole->createDocument('rules', $domainDocument);
+
+                        Console::info('Issuing a TLS certificate for the main domain (' . $domain->get() . ') in a few seconds...');
+
+                        (new Certificate())
+                            ->setDomain($domainDocument)
+                            ->setSkipRenewCheck(true)
+                            ->trigger();
+                    }
+                }
+                $domains[$domain->get()] = true;
+
+                Authorization::reset(); // ensure authorization is re-enabled
+            }
+            Config::setParam('domains', $domains);
```

Comment:
im not sure this works as intended when we have multiple workers, as each worker holds its own copy of config in its memory.

Reply:
I dont think config behaves as a static variable